weave/flow/eval.py

```diff
@@ -248,6 +248,9 @@ async def predict_and_score(
                     """
                 )
                 raise OpCallError(message)
+            # this is a catchall in case the user's scorer function throws
+            except Exception as e:
```

Comment:
This makes debugging really bad. Tracebacks will no longer be printed to the console.

Reply:
What would you do differently?

Some alternatives:
1. Add a debug flag to let the user decide if they want the eval to raise (will stop the eval)
2. Save the whole traceback instead of just `str(e)`

Reply:
We need to at least print the whole traceback to the console

Reply:
It would be super noisy if you had an exception in a scorer function and had many examples though.

Imagine a 100 line trace * 100 examples -- very noisy and could lag if you print to console!

Reply:
You have to show developers tracebacks in the console if you want them to be able to fix them.

I tried this before with swallowing exceptions. It sucks. I need to know the exact traceback and line of code a problem came from, when I'm developing a scorer.

Try developing your own scorer function without it!

Reply:
But you will see the traceback in the UI...

Let's lay out a few things that we certainly want and don't want to narrow the solution space:
1. The user definitely needs to see a traceback somewhere to figure out how to fix their error.
2. We definitely can't print a large volume of tracebacks to the console because it will be extremely noisy and also slow down the user's code (try printing 10k, 100k lines to term)

Some solutions:
1. Current option -- user sees tracebacks in the UI only (not console).  Since we don't print to console, there is no possibility of program slowdown
2. Current option + capture exceptions and raise them in python at the end of evaluation